### PR TITLE
fixed Error when trying to set boolean field to `false` on google_data_catalog_tag

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/7097

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
datacatalog: fixed error when trying to set `bool_value` to false on `google_data_catalog_tag`
```

### CODE REVIEW

File: third_party/terraform/utils/transport.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func isEmptyValue(v reflect.Value) bool {
 	case reflect.Array, reflect.Map, reflect.Slice, reflect.String:
 		return v.Len() == 0
 	case reflect.Bool:
-		return !v.Bool()
+		return false
```
Comment (danawillow) [inline third_party/terraform/utils/transport.go:29]: This doesn't make sense to me. It seems to be saying that booleans can _never_ be an empty value, which isn't true. False is the empty value for booleans, just like 0 is for ints and "" is for strings. I think what you actually want to do is set `send_empty_value: true` on the field in question.

Comment (edwardmedia) [reply third_party/terraform/utils/transport.go:29]: To cast a empty string to boolean, yes, it is false. But a false value of boolean type is concrete and clear, and it should be treated the same as true. But after comparing with values of other types, like treating Integer number 0 as `empty` from this function, I think my initial interpretation of the purpose of `isEmptyValue()` was different.

File: products/datacatalog/terraform.yaml [hunk 2]
```diff
@@ -155,6 +155,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           force_delete: "true"
         oics_vars_overrides:
           force_delete: "true"
+      - !ruby/object:Provider::Terraform::Examples
```
Comment (danawillow) [inline products/datacatalog/terraform.yaml:158]: Do you think this example is unique enough that it should be in the documentation? If not, I'd recommend doing `skip_docs` for it.

**Review by danawillow — COMMENTED** [review]
I updated the title of this PR to describe what it actually does now, and the changelog note to match the style guide. Please take a look at the changes I made so you can see what we'd like to see in those two places :)

File: products/datacatalog/terraform.yaml (hunk 2 above)
Comment (edwardmedia) [reply products/datacatalog/terraform.yaml:158]: Agreed, it is almost identical to one existing case and omitting it

**Comment by edwardmedia** [comment]
`tpg-vcr-test` failed. Likely it is `timeout` issue like the other case for google_cloudbuild_trigger

**Review by danawillow — APPROVED** [review]

**Comment by ndmckinley** [comment]
This PR seems to have broken `TestAccDataCatalogTag_*`.

**Comment by ndmckinley** [comment]
Yes, confirmed - with this PR, all creates of these resources fail because it sends `source.0.bool_value` unconditionally, but only one field under `source` can be sent.  Reverting - we'll need to find another way to address https://github.com/hashicorp/terraform-provider-google/issues/7097.  Maybe an expander?

**Comment by edwardmedia** [comment]
I see what happens. Maybe we could reconsider updating ` isEmptyValue()` in transport.go?

```
{
 "column": "first_name",
 "fields": {
  "pii_type": {
   "boolValue": false,
   "enumValue": {
    "displayName": "NONE"
   }
  },
  "source": {
   "boolValue": false,
   "stringValue": "my-string"
  }
 },
 "template": "projects/myproject/locations/us-central1/tagTemplates/tf_test_my_templated2jfxsfx7o"
}
```

**Comment by ndmckinley** [comment]
Yeah, this will require a bit of special handling - we want to send the empty value if it is _set_ in config, which we can only learn from `d.GetOkExists()`, which will require reworking of the template or some new kinds of custom code.

**Comment by edwardmedia** [comment]
One workaround for this issue is to override `expandNestedDataCatalogTagFields()` to make sure only one `value` will be assigned. To achieve this, shift the order of `value` assignments and put the `boolValue` to be the last. If any `value` assignment occurs before `boolValue` is hit, skip `boolValue` assignment.

**Comment by ndmckinley** [comment]
That's also a good idea.  That could probably be done in an encoder.

**Comment by edwardmedia** [comment]
Implemented the fix using *encoder* https://github.com/GoogleCloudPlatform/magic-modules/pull/4004

**Pull request merged**